x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/sections/action_connector_form/connector_edit_flyout.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { useContext, useCallback } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiTitle,
+  EuiFlyoutHeader,
+  EuiFlyout,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiIcon,
+} from '@elastic/eui';
+import { ActionsConnectorsContext } from '../../context/actions_connectors_context';
+import { ActionConnectorForm } from './action_connector_form';
+import { useAppDependencies } from '../../app_dependencies';
+import { ActionConnectorTableItem } from '../../../types';
+
+export interface ConnectorEditProps {
+  connector: ActionConnectorTableItem;
+}
+
+export const ConnectorEditFlyout = ({ connector }: ConnectorEditProps) => {
+  const { actionTypeRegistry } = useAppDependencies();
+  const { editFlyoutVisible, setEditFlyoutVisibility } = useContext(ActionsConnectorsContext);
+  const closeFlyout = useCallback(() => setEditFlyoutVisibility(false), [setEditFlyoutVisibility]);
+
+  if (!editFlyoutVisible) {
+    return null;
+  }
+
+  const actionTypeModel = actionTypeRegistry.get(connector.actionTypeId);
+
+  return (
+    <EuiFlyout onClose={closeFlyout} aria-labelledby="flyoutActionAddTitle" size="m">
+      <EuiFlyoutHeader hasBorder>
+        <EuiFlexGroup gutterSize="s" alignItems="center">
+          {actionTypeModel ? (
+            <EuiFlexItem grow={false}>
+              <EuiIcon type={actionTypeModel.iconClass} size="m" />
+            </EuiFlexItem>
+          ) : null}
+          <EuiFlexItem>
+            <EuiTitle size="s">
+              <h3 id="flyoutTitle">
+                <FormattedMessage
+                  defaultMessage="Edit connector"
```

Comment:
Should we change the label when a readonly user sees the connector details? Maybe a "Close" button instead of "Save" as well?

Reply:
Save button is not visible in case of readonly users, but the other fields are enabled for editing and the title still says "Edit connector". It is a question - how this edit form should behave for readonly users? Do we really need to show it (in case is the user wants to see more details about connector)? Maybe it can be text only Connector Details view?

Reply:
Form fields to be disabled with a message at the top explaining why.